nilearn/image/resampling.py

```diff
@@ -103,25 +104,43 @@ def coord_transform(x, y, z, affine):
 
         Returns
         -------
-        x : number or ndarray
+        x : number or ndarray (same shape as input)
             The x coordinates in the output space
-        y : number or ndarray
+        y : number or ndarray (same shape as input)
             The y coordinates in the output space
-        z : number or ndarray
+        z : number or ndarray (same shape as input)
             The z coordinates in the output space
 
         Warning: The x, y and z have their Talairach ordering, not 3D
-        numy image ordering.
+        numpy image ordering.
+
+    Examples
+    --------
+    Transform data from coordinates to brain space. The "affine" matrix
+    can be found as the ".affine" attribute of a nifti image, or using
+    the "get_affine()" method for older nibabel installations::
+
+        >>> from nilearn import datasets, image
+        >>> niimg = datasets.load_mni152_template()
+        >>> # Find the MNI coordinates of the voxel (10, 10, 10)
+        >>> image.coord_transform(50, 50, 50, niimg.affine)
+        (-10.0, -26.0, 28.0)
+
     """
     squeeze = (not hasattr(x, '__iter__'))
+    return_number = isinstance(x, numbers.Number)
+    x = np.asanyarray(x)
+    shape = x.shape
     coords = np.c_[np.atleast_1d(x).flat,
                    np.atleast_1d(y).flat,
                    np.atleast_1d(z).flat,
                    np.ones_like(np.atleast_1d(z).flat)].T
     x, y, z, _ = np.dot(affine, coords)
+    if return_number:
+        return x.item(), y.item(), z.item()
```

Comment:
Is there a unit test covering that case ?

Reply:
There is already in test_coord_transform module testing single integer as inputs and return output the single integer.